overlord/servicestate/quota_control.go

```diff
@@ -179,14 +179,38 @@ func RemoveQuota(st *state.State, name string) error {
 			return fmt.Errorf("internal error: parent group of %q, %q not found", name, grp.ParentGroup)
 		}
 
-		newSubgroups := make([]string, 0, len(parent.SubGroups)-1)
-		for _, sub := range parent.SubGroups {
-			if sub != name {
-				newSubgroups = append(newSubgroups, sub)
+		errParentNoRefChildMsg := fmt.Errorf("internal error: parent group of %q, %q does not reference %q", name, parent.Name, name)
```

Comment:
this feels more like something we should log about than error if we care,  we are not going to produce an inconsistent state afaict, we are just noticing that one was there (which shouldn't happen because of the other checks, but either way)

Reply:
mmm I suppose it does make sense that of all the methods RemoveQuota should be the most accommodating when it comes to inconsistent state like this, especially considering that removing the quota group here when the parent has no reference to it will actually make the state more consistent than it was before possibly fixing the state of the system. I'll fix this up a bit, but it will require more tests I think to validate it works as advertised then

Reply:
as discussed on IRC actually none of this is possible right now, because these error cases are about inconsistent state, but the state we use here was already validated by AllQuotas, so we can't have these inconsistent state situations here.

we may one day want to be able to remove groups from inconsistent state (possibly as a corrective measure to fix the state), but that requires a bit more work than we want for this PR, so I will just remove the unnecessary error handling code and then this PR can go in